### ISSUE TITLE
Coq 8.11 degradation. Anomaly "Not enough arguments."

### ISSUE THREAD

**Issue by k32** [coq/coq#12740]
#### Description of the problem
The following snippet of code causes coqchk to fail with fatal error: `Anomaly "Not enough arguments."` The same code passes coqchk on 8.9.

```coq
From Coq Require Import
     List.

Import ListNotations.

Class StateSpace (S TE : Type) :=
  { rule : S -> S -> TE -> Prop;
  }.

Section defn.
  Context {S TE} `{StateSpace S TE}.
  Let T := list TE.

  Definition TraceEnsemble := T -> Prop.

  Inductive TraceEnsembleConcat (e1 e2 : TraceEnsemble) : TraceEnsemble :=
  | et_concat : forall t1 t2, e1 t1 -> e2 t2 -> TraceEnsembleConcat e1 e2 (t1 ++ t2).
End defn.
```

Rearranging definition like this: `Definition TraceEnsemble := list TE -> Prop.` seemingly makes the problem disappear

#### Coq Version

```
$ coqtop -v
The Coq Proof Assistant, version 8.11.2 (July 2020)
compiled on Jul 22 2020 22:31:00 with OCaml 4.08.1
$ uname -a
Linux 5.4.0-42-generic #46-Ubuntu SMP x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by ejgallego** [comment]
Seem fixed in 8.12.

**Comment by k32** [comment]
Great news, thanks. I didn't find this issue in search.
